core/state_transition.go

```diff
@@ -307,9 +308,17 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	homestead := st.evm.ChainConfig().IsHomestead(st.evm.Context.BlockNumber)
 	istanbul := st.evm.ChainConfig().IsIstanbul(st.evm.Context.BlockNumber)
 	subnetEVM := st.evm.ChainConfig().IsSubnetEVM(st.evm.Context.Time)
+	isTxAllowList := st.evm.ChainConfig().IsTxAllowList(st.evm.Context.Time)
 
 	contractCreation := msg.To() == nil
 
+	if isTxAllowList {
+		txAllowListRole := precompile.GetTxAllowListStatus(st.state, sender.Address())
```

Comment:
In the `contractDeployerAllowList` we checked tx.Origin. Should we check it here as well instead of sender?

Reply:
is `sender.Address()` equivalent to `msg.sender`? If I understand correctly, tx.origin will always return the EOA that originated the TX and sender.Address will change to a contract if there is a nested call. We probably want to only whitelist EOAs, right? Don't want to need to make people also verify contracts? I think there a couple tests that might need modification if we switch, but we should be clear about intended behavior.

Reply:
tx.Origin is the name of the variable on the `txContext` within the EVM, so for the contract deployer list, we check that since it is what is available. The `sender.Address()` is the same address as Connor said: the EOA address that originates the transaction.

The txContext is available here as well, but it seems more direct to access it in this case through `sender.Address()`, so I think we can leave this as is.

Reply:
When I moved to precheck, I switched to st.msg.From() since that's what the rest of the function used. I think that's acceptable?

Reply:
I was just under impression `sender.Address()` was equal to `msg.sender` but seems it was an EOA. it actually verifies that under: 
https://github.com/ava-labs/subnet-evm/blob/f98dcad25f9f2ca08f92e5b50e11a1e19b4f0392/core/state_transition.go#L241-L245
Thanks for clarification!